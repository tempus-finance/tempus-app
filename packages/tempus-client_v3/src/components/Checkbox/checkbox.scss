.tc__checkbox {
  display: flex;
  align-items: center;
}

.tc__checkbox__box-container {
  width: 20px;
  height: 20px;
  border: 2px solid var(--primaryMain);
  padding: 2px;
  display: inline-flex;
  align-items: center;
  position: relative;
  box-sizing: border-box;

  input[type='checkbox'] {
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    opacity: 0;
    position: absolute;
    z-index: 1;
  }
}

.tc__checkbox__checked-box {
  width: 100%;
  height: 100%;
  background: var(--primaryMain);
}

.tc__checkbox__label {
  padding-left: 6px;
}
