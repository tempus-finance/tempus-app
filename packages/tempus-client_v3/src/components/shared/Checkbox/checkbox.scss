@import '../common.scss';

@include checked-input('checkbox');
