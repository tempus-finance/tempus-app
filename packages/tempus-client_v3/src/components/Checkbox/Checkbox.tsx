import { Typography } from '../shared';
import './checkbox.scss';

interface CheckboxProps {
  id: string;
  checked: boolean;
  label?: string;
}

const Checkbox = (props: CheckboxProps & React.HTMLProps<HTMLInputElement>) => {
  const { id, checked, label, ...checkboxProps } = props;

  return (
    <span className="tc__checkbox">
      <span className="tc__checkbox__box-container">
        <input {...checkboxProps} id={id} checked={checked} type="checkbox" />
        <span className={`${checked ? 'tc__checkbox__checked-box' : ''}`} aria-hidden="true"></span>
      </span>
      {label && (
        <label htmlFor={id}>
          <Typography
            variant="body-secondary"
            className={`tc__checkbox__label ${checked ? 'tc__checkbox__label-selected' : ''}`}
          >
            {label}
          </Typography>
        </label>
      )}
    </span>
  );
};

export default Checkbox;
