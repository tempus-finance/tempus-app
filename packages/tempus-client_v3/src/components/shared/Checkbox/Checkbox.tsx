import { ChangeEvent, FC, useMemo } from 'react';
import Typography from '../Typography';
import './checkbox.scss';

export interface CheckboxProps {
  checked: boolean;
  label?: string;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
}

let idCounter = 0;

const Checkbox: FC<CheckboxProps> = props => {
  const { checked, label, onChange } = props;
  const id = useMemo(() => `checkbox-${idCounter++}`, []);

  return (
    <span className="tc__checkbox">
      <span className="tc__checkbox__box-container">
        <input id={id} checked={checked} type="checkbox" onChange={onChange} />
        <span className={`${checked ? 'tc__checkbox__checked-box' : ''}`} aria-hidden="true"></span>
      </span>
      {label && (
        <label htmlFor={id} className="tc__checkbox__label">
          <Typography variant="body-secondary" weight={checked ? 'bold' : 'regular'}>
            {label}
          </Typography>
        </label>
      )}
    </span>
  );
};

export default Checkbox;
