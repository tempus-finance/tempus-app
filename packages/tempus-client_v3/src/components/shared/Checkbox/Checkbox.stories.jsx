import React from 'react';

import Checkbox from './Checkbox';

export default {
  title: 'Checkbox',
  component: Checkbox,
  argTypes: {},
};

const style = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  padding: '10px',
};

const Template = args => (
  <div style={style}>
    <Checkbox {...args} />
  </div>
);

export const Unchecked = Template.bind({});
Unchecked.args = {
  checked: false,
  label: 'unchecked',
};

export const Checked = Template.bind({});
Checked.args = {
  checked: true,
  label: 'checked',
};
