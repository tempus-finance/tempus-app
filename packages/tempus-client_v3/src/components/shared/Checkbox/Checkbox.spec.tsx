import { fireEvent, render } from '@testing-library/react';
import Checkbox, { CheckboxProps } from './Checkbox';

const mockOnChange = jest.fn();

const defaultProps: CheckboxProps = {
  checked: false,
  label: 'myCheckbox',
  onChange: mockOnChange,
};

const subject = (props: CheckboxProps) => render(<Checkbox {...props} />);

describe('Checkbox', () => {
  it('renders an unchecked checkbox', () => {
    const { queryByText, getByRole } = subject(defaultProps);

    const actualText = queryByText('myCheckbox');
    const actualCheckbox = getByRole('checkbox');

    expect(actualText).not.toBeNull();
    expect(actualCheckbox).not.toBeNull();
    expect(actualCheckbox).not.toBeChecked();

    expect(actualCheckbox).toMatchSnapshot();
  });

  it('renders an checked checkbox', () => {
    const { queryByText, getByRole } = subject({ ...defaultProps, checked: true });

    const actualText = queryByText('myCheckbox');
    const actualCheckbox = getByRole('checkbox');

    expect(actualText).not.toBeNull();
    expect(actualCheckbox).not.toBeNull();
    expect(actualCheckbox).toBeChecked();

    expect(actualCheckbox).toMatchSnapshot();
  });

  it('calls `onChange` when the checkbox is clicked', () => {
    const { queryByText, getByRole } = subject({ ...defaultProps });

    const actualText = queryByText('myCheckbox');
    const actualCheckbox = getByRole('checkbox');

    expect(actualText).not.toBeNull();
    expect(actualCheckbox).not.toBeNull();
    expect(actualCheckbox).not.toBeChecked();

    fireEvent.click(actualCheckbox);

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
