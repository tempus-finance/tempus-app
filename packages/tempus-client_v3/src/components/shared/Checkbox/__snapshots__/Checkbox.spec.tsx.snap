// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders an checked checkbox 1`] = `
<input
  checked=""
  id="checkbox-2"
  type="checkbox"
/>
`;

exports[`Checkbox renders an unchecked checkbox 1`] = `
<input
  id="checkbox-1"
  type="checkbox"
/>
`;
